test/suites/common/xcm/test-mock-hrmp-transact.ts

```diff
@@ -0,0 +1,204 @@
+    });
+
+    it({
+      id: "T01",
+      title: "Should succeed using sovereign account from signed origin",
+      test: async function () {
+        // Generate random receiver address
+        let random: KeyringPair;
+        random = chain == 'frontier-template' ? generateKeyringPair() : generateKeyringPair("sr25519");
+
+        // Get Pallet balances index
+        const metadata = await polkadotJs.rpc.state.getMetadata();
+        const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+        (pallet) => {
+            return pallet.name === "Balances";
+        }
+        ).index;
+
+        const transferCall = polkadotJs.tx.balances.transfer(
+        random.address,
+        transferredBalance / 10n
+        );
+        const transferCallEncoded = transferCall?.method.toHex();
+
+        // We are going to test that we can receive a transact operation from parachain 1
+        // using descendOrigin first
+        const xcmMessage = new XcmFragment({
+            assets: [
+                {
+                multilocation: {
+                    parents: 0,
+                    interior: {
+                    X1: { PalletInstance: balancesPalletIndex },
+                    },
+                },
+                fungible: transferredBalance / 4n,
+                },
+            ],
+            descend_origin: sendingAddress,
+        })
+        .descend_origin()
+        .withdraw_asset()
+        .buy_execution()
+        .push_any({
+            Transact: {
+            originType: "SovereignAccount",
+            requireWeightAtMost: new BN(1000000000),
+            call: {
+                encoded: transferCallEncoded,
+            },
+            },
+        })
+        .as_v2();
+
+
+        // Send an XCM and create block to execute it
+        await injectHrmpMessageAndSeal(context, 1, {
+            type: "XcmVersionedXcm",
+            payload: xcmMessage,
+        } as RawXcmMessage);
+
+        // Make sure the state has ALITH's foreign parachain tokens
+        const testAccountBalance = (
+        await polkadotJs.query.system.account(random.address)
+        ).data.free.toBigInt();
+
+        expect(testAccountBalance).to.eq(transferredBalance / 10n);
+        },
+    });
+
+    it({
+        id: "T02",
+        title: "Should succeed using sovereign account from root origin",
+        test: async function () {
+            // Generate random receiver address
+            let random: KeyringPair;
+            random = chain == 'frontier-template' ? generateKeyringPair() : generateKeyringPair("sr25519");
+
+            // Get Pallet balances index
+            const metadata = await polkadotJs.rpc.state.getMetadata();
+            const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+            (pallet) => {
+                return pallet.name === "Balances";
+            }
+            ).index;
```

Comment:
This is a cleaner way to get pallet indices.
```suggestion
const balancesPalletIndex = metadata.asLatest.pallets
          .find(({ name }) => name.toString() == "Balances")!
          .index.toNumber();
```